huaweicloud/services/cae/resource_huaweicloud_cae_compontent.go

```diff
@@ -0,0 +1,535 @@
+		return diag.Errorf("error creating CAE component : %s", err)
+	}
+
+	createRespBody, err := utils.FlattenResponse(createResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("metadata.id", createRespBody)
+	if err != nil {
+		return diag.Errorf("error creating CAE component: ID is not found in API response")
+	}
+
+	d.SetId(id.(string))
+	return resourceComponentRead(ctx, d, meta)
+}
+
+// GetCompontentById is a method to query component details from a specified application ID using given parameters.
+func GetCompontentById(cfg *config.Config, region, applicationId, id string) (interface{}, error) {
+	var (
+		httpUrl     = "v1/{project_id}/cae/applications/{application_id}/components/{component_id}"
+		product     = "cae"
+		client, err = cfg.NewServiceClient(product, region)
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error creating CAE Client: %s", err)
+	}
+
+	getPath := client.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath = strings.ReplaceAll(getPath, "{application_id}", applicationId)
+	getPath = strings.ReplaceAll(getPath, "{component_id}", id)
+	getComponentOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	resp, err := client.Request("GET", getPath, &getComponentOpt)
+	if err != nil {
+		return nil, parseComponentResponseError(err)
+	}
+
+	getComponentRespBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return getComponentRespBody, nil
+}
+
+func resourceComponentRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	componentRespBody, err := GetCompontentById(cfg, region, d.Get("application_id").(string), d.Id())
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving CAE component")
+	}
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("environment_id", utils.PathSearch("spec.env_id", componentRespBody, nil)),
+		d.Set("metadata", flattenMetadata(d.Get("metadata.0.annotations"), utils.PathSearch("metadata", componentRespBody, nil))),
+		d.Set("spec", flattenSpec(d.Get("spec.0.build.0.parameters"), utils.PathSearch("spec", componentRespBody, nil))),
+		d.Set("created_at", utils.PathSearch("metadata.created_at", componentRespBody, nil)),
+		d.Set("updated_at", utils.PathSearch("metadata.updated_at", componentRespBody, nil)),
+	)
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func parseComponentResponseError(respErr error) error {
+	var err interface{}
+	if errCode, ok := respErr.(golangsdk.ErrDefault400); ok {
+		pErr := json.Unmarshal(errCode.Body, &err)
+		if pErr != nil {
+			return pErr
+		}
+
+		errCode, err := jmespath.Search("error_code", err)
+		if err != nil {
+			return fmt.Errorf("error parse errorCode from response body: %s", err.Error())
+		}
+
+		if errCode == "CAE.01500404" || errCode == "CAE.01500208" {
+			return golangsdk.ErrDefault404{}
+		}
```

Comment:
```suggestion
		// Error code FSS.0500 indicates that the function to which the trigger belongs has been deleted.
		if errCode == "CAE.01500404" || errCode == "CAE.01500208" {
			return golangsdk.ErrDefault404(errCode)
		}
```

Reply:
```
// ParseQueryError400 is a method used to parse whether a 404 error message means the resources not found.
// For the CAE service, there are some known 404 error codes:
// + CAE.01500208: application or component does not found.
// + CAE.01500404: environment does not found.
func ParseQueryError400(err error, specErrors []string) error {
	var err400 golangsdk.ErrDefault400
	if errors.As(err, &err400) {
		var apiError interface{}
		if jsonErr := json.Unmarshal(err400.Body, &apiError); jsonErr != nil {
			return err
		}

		errCode, err := jmespath.Search("error_code", apiError)
		if errCode != nil {
			return err
		}

		for _, v := range specErrors {
			if errCode == v {
				return golangsdk.ErrDefault404{}
			}
		}
	}
	return err
}
```
It is recommend to define this function in the `common.go` for CAE service package.

Reply:
fixed